servers/zms/src/main/java/com/yahoo/athenz/zms/messaging/DomainChangeMessageGenerator.java

```diff
@@ -0,0 +1,132 @@
+package com.yahoo.athenz.zms.messaging;
+
+import com.yahoo.athenz.common.messaging.DomainChangeMessage;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DomainChangeMessageGenerator {
```

Comment:
I'm not sure if I like the functionality in this class. Nowhere in the code we look at the URIs and make assumptions about the components and here looks like we're parsing and handling accordingly. This could lead to incorrect events possibly being generated. Also at this point, we have really verified and/or normalized the arguments so we don't want to base our events on not-normalized argument values.

I'd prefer that in each implementation method we actually create the event object since we're guaranteed to know exactly what the the event is.